test/tests/functional/pbs_schedule_indirect_resources.py

```diff
@@ -0,0 +1,128 @@
+        self.server.add_resource(res, res_type, flag)
+
+        # Add resource to the resources line in sched_config
+        self.scheduler.add_resource(res)
+
+        # Set resource as the node_group_key
+        attr = {'node_group_enable': 'True', 'node_group_key': res}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+
+    def submit_job(self, attr):
+        """
+        Helper function to submit a sleep job with provided attributes
+        """
+        job = Job(TEST_USER1, attr)
+        job.set_sleep_time(3600)
+        jobid = self.server.submit(job)
+
+        return (jobid, job)
+
+    def test_node_grouping_with_indirect_res(self):
+        """
+        Test node grouping with indirect resources set on some nodes
+        Steps:
+        -> Configure the system to have 6 vnodes with custom resource 'foostr'
+        set as the node_group_key
+        -> Set 'foostr' to 'A', 'B', and 'C' on the first three vnodes
+        -> Set 'foostr' for the next three vnodes to point to the first three
+        vnodes correspondingly
+        -> Verify that the last three vnodes are part of their respective
+        placement sets
+        """
+
+        # Delete any existing vnodes on the system
+        self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+
+        # Create 6 vnodes
+        attr = {'resources_available.ncpus': 1}
+        self.server.create_vnodes('vnode', attr, 6, mom=self.mom)
+
+        # Configure a system with 6 vnodes and 'foostr' as node_group_key
+        self.config_complex_for_grouping('foostr')
+
+        # Set 'foostr' to 'A', 'B' and 'C' respectively for the first
+        # three vnodes
+        attr = {'resources_available.foostr': 'A'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[0]')
+        attr = {'resources_available.foostr': 'B'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[1]')
+        attr = {'resources_available.foostr': 'C'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[2]')
+
+        # Set 'foostr' for last three vnodes as indirect resource to
+        # the first three vnodes correcpondingly
+        attr = {'resources_available.foostr': '@vnode[0]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[3]')
+        attr = {'resources_available.foostr': '@vnode[1]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[4]')
+        attr = {'resources_available.foostr': '@vnode[2]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[5]')
+
+        # Verify that the last three vnodes are part of their respective
+        # placement sets
+        # Submit 3 jobs requesting placement sets 'A', 'B' and 'C', and vnodes
+        # vnode[3], vnode[4] and vnode[5] respectively
+        attr1 = {'Resource_List.select': '1:foostr=A:vnode=vnode[3]'}
+        attr2 = {'Resource_List.select': '1:foostr=B:vnode=vnode[4]'}
+        attr3 = {'Resource_List.select': '1:foostr=C:vnode=vnode[5]'}
```

Comment:
This isn't really testing node grouping.  It is testing indirect resources.  If you want to test node grouping, you can't request the node grouping resource.  Also requesting a specific vnode won't help either.  You need to submit 3 jobs requesting 2 nodes.  Make sure they're on the correct set of vnodes.

Reply:
Thanks for this comment, It made me find another problem in the code where I was missing the change.